README.rst

```diff
@@ -225,7 +229,8 @@ Compatibility with Python and GMT versions
 ======= ========== =========
 PyGMT   GMT        Python
 ======= ========== =========
-0.2.1   >=6.1.1    >=3.6
+0.3.0   >=6.1.1    >=3.6
+0.2.1   >=6.1.1    3.6 - 3.9
```

Comment:
1. See https://github.com/GenericMappingTools/pygmt/pull/763#discussion_r549190174 for why we set `>=3.6` for PyGMT v0.2.1
2. The [minimum required Python version is 3.7](https://github.com/GenericMappingTools/pygmt/blob/02858eb5f916d3071ab2b104e41e4e8251896efb/setup.py#L30) for PyGMT v0.3.0
```suggestion
0.3.0   >=6.1.1    >=3.7
0.2.1   >=6.1.1    >=3.6
```